enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,133 @@
+---
+title: improved-platform-tests
+authors:
+  - "@soltysh"
+reviewers:
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+api-approvers:
+  - None
+creation-date: 2022-11-16
+last-updated: 2022-11-28
+tracking-link:
+  - None
+replaces:
+  - "enhancements/testing/extended-platform-tests.md"
+---
+
+# Improved Platform Tests
+
+## Summary
+
+Separate OpenShift tests from the ones coming from kubernetes. By doing so we will:
+- shorten the time when we start running most up-to-date tests from kubernetes;
+- run all tests coming from kubernetes, especially those introduced in patch releases;
+- remove the necessity to vendor kubernetes tests directly;
+- simplify the [kubernetes update process](https://github.com/openshift/kubernetes/blob/master/REBASE.openshift.md).
+
+## Motivation
+
+Over the past three releases the time to deliver the [updated kubernetes (o/k)](https://github.com/openshift/kubernetes/)
+to our [tests (origin)](https://github.com/openshift/origin/) has increased significantly.
+The delta for these was:
+- OpenShift 4.10 / kubernetes 1.23 - 7 days (compare [o/k#1087](https://github.com/openshift/kubernetes/pull/1087)
+  vs [origin#26753](https://github.com/openshift/origin/pull/26753)).
+- OpenShift 4.11 / kubernetes 1.24 - 21 days (compare [o/k#1252](https://github.com/openshift/kubernetes/pull/1252)
+  vs [origin#27244](https://github.com/openshift/origin/pull/27244)).
+- OpenShift 4.12 / kubernetes 1.25 - 40 days (compare [o/k#1360](https://github.com/openshift/kubernetes/pull/1360)
+  vs [origin#27526](https://github.com/openshift/origin/pull/27526)).
+
+Moreover, in the past two releases the changes to origin repository were reverted
+at least once, due to multiple issues found, which also affected the increased
+delays.
+
+### User Stories
+
+- As an OpenShift developer, I want to be able to seamlessly run OpenShift
+  and kubernetes-specific tests using `openshift-tests run`.
+- As an OpenShift developer, I want to be able to run newest kubernetes tests
+  without the necessity to bump vendored code.
+- As an OpenShift developer, I want to be able to extend and monitor cluster
+  during tests execution.
+- As an OpenShift user, I want to be able to verify that my cluster is providing
+  the desired functionality running one of the provided test suites.
+
+### Goals
+
+1. Reduce the risk updating kubernetes caused by excessive co-located code.
+2. Minimize time when we start running newest kubernetes tests.
+3. Maintain `openshift-tests` API for running tests.
+
+### Non-Goals
+
+1. Abandon existing tests.
+2. Change the available functionality of `openshift-tests`.
+3. Minimize memory footprint for running `openshift-tests`.
```

Comment:
4. moving the upgrade+monitoring tests that live in openshift-tests itself, out of that binary.

Reply:
Can you elaborate on this one @bparees.

Reply:
The code for the openshift-tests binary defines some "virtual" tests (they aren't ginkgo tests, they are tests that are run/managed in code and directly spit out junit results, iirc).  Those tests need to be part of the wrapper openshift-tests binary (vs just being another e2e test/binary that gets invoked/wrapped) because they either:

1) need to run during the entire duration of the test job (monitoring the cluster as other tests are run)
2) orchestrate upgrades and specific tests/verifications performed at various points before/during/after the upgrade

So given the way those tests need to be orchestrated/executed/managed they don't fit as a traditional e2e test that is just part of a ginkgo suite.  

That is why I wanted to call out that it is a non-goal to move those out of the openshift-tests wrapper.  They will need to stay where they are (or if they're going to move, that's going to be a different conversation about how to move them and where to move them to, and probably not something we want to hold this EP up over)

Reply:
marking this resolved since i see it was added as item 4.

Reply:
It's a non-goal, but I've added this and the split of k8s tests as "Even more future work"